internal/serving/testing/minimatch_test.go

```diff
@@ -39,15 +42,68 @@ func TestNewMiniMatch(t *goTesting.T) {
 		t.Errorf("could not start Mini Match %s", err)
 	}
 	defer mm.Stop()
-	feClient, err := mm.GetFrontendClient()
-	if err != nil {
-		t.Errorf("could not get frontend client %s", err)
-	}
-	result, err := feClient.CreatePlayer(context.Background(), &pb.CreatePlayerRequest{})
-	if err != nil {
-		t.Errorf("could not start Mini Match %s", err)
+
+	t.Run("FrontendClient Test", func(t *goTesting.T) {
+		feClient, err := mm.GetFrontendClient()
+		if err != nil {
+			t.Errorf("could not get frontend client %s", err)
+		}
+		result, err := feClient.CreatePlayer(context.Background(), &pb.CreatePlayerRequest{})
+		if err != nil {
+			t.Errorf("could not start Mini Match %s", err)
+		}
+		if result == nil {
+			t.Errorf("insert player was not successful %v", result)
+		}
+	})
+
+	testStubs := []struct {
+		method   string
+		endpoint string
+		response string
+	}{
+		// TODO: Diagnose why the response is 
+		//	{code: 1, error: "grpc: the client connection is closing"}
+		// {
+		// 	method:   "GET",
+		//	endpoint: "v1/frontend/players/123",
+		//	response: "Not Found",
+		//},
+		{
+			method:   "GET",
+			endpoint: "nowhere",
+			response: "Not Found\n",
+		},
+		{
+			method:   "GET",
+			endpoint: "ping",
+			response: "pong",
+		},
 	}
-	if result == nil {
-		t.Errorf("insert player was not successful %v", result)
+
+	for _, stub := range testStubs {
+		t.Run(fmt.Sprintf("ProxyTest-%s-%s", stub.method, stub.endpoint), func(t *goTesting.T) {
+			apiAddr := fmt.Sprintf("http://:%d/%s", mm.Config.GetInt("api.frontend.proxyport"), stub.endpoint)
```

Comment:
It's better to return an http.Client along with a base URL. The reason why is because in a later change we'll be providing TLS support so the http.Client will need to be authenticated first before we can use it and it'd be best not have the tests be aware of that.